src/core/obj.js

```diff
@@ -1147,8 +1148,33 @@ var XRef = (function XRefClosure() {
               uncompressed: true,
             };
           }
-          var contentLength = skipUntil(buffer, position, endobjBytes) + 7;
-          var content = buffer.subarray(position, position + contentLength);
+          let contentLength, startPos = position + token.length;
+
+          // Find the next "obj" string, rather than "endobj", to ensure that
+          // we won't skip over a new 'obj' operator in corrupt files where
+          // 'endobj' operators are missing (fixes issue9105_reduced.pdf).
+          while (startPos < buffer.length) {
+            let endPos = startPos + skipUntil(buffer, startPos, objBytes) + 4;
+            contentLength = endPos - position;
+
+            let tokenStr = bytesToString(buffer.subarray(endPos - 8, endPos));
```

Comment:
What does this number 8 stand for? Can we make a constant out of it or add a comment?

Reply:
> What does this number 8 stand for?

It's just to get the end of the stream such that we can check that if e.g. "&nbsp;endobj&nbsp;" was found; i.e. the endobj operator.